scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/SoftKeywords.scala

```diff
@@ -0,0 +1,87 @@
+package scala.meta.internal.parsers
+
+import scala.meta.Dialect
+import scala.meta.tokens.Token
+import scala.meta.tokens.Token.Ident
+import scala.meta.internal.classifiers.classifier
+
+class SoftKeywords(dialect: Dialect) {
+
+  def isIdentAnd(token: Token, pred: String => Boolean): Boolean = token match {
```

Comment:
Make private?

Reply:
actually, this duplicates the method in `ScalametaParser` above. should we do this instead:

```
private[parsers] object ScalametaParser {
  def isIdentAnd(token: Token, pred: String => Boolean): Boolean = token match {
    case Ident(value) => pred(value.stripPrefix("`").stripSuffix("`"))
    case _ => false
  }
}
```
and then import it in both classes?

Reply:
Moved it actually to ScalametParserObject and is now reused in SoftKeywords and ScalametaParser.